Tests/MapboxMapsTests/Foundation/MapViewTests.swift

```diff
@@ -261,109 +261,112 @@ final class MapViewTests: XCTestCase {
+    var mapView: MapView!
+    var window: UIWindow!
+    var metalView: MockMetalView!
+
+    override func setUpWithError() throws {
+        try super.setUpWithError()
+        notificationCenter = MockNotificationCenter()
+        bundle = MockBundle()
+        bundle.infoDictionaryStub.defaultReturnValue = ["UIApplicationSceneManifest": []]
+        cameraAnimatorsRunnerEnablable = MockMutableEnablable()
+        displayLink = MockDisplayLink()
+        locationProducer = MockLocationProducer()
+        dependencyProvider = MockMapViewDependencyProvider()
+        dependencyProvider.notificationCenter = notificationCenter
+        dependencyProvider.bundle = bundle
+        dependencyProvider.cameraAnimatorsRunnerEnablable = cameraAnimatorsRunnerEnablable
+        dependencyProvider.makeDisplayLinkStub.defaultReturnValue = displayLink
+        dependencyProvider.makeLocationProducerStub.defaultReturnValue = locationProducer
+        orientationProvider = MockInterfaceOrientationProvider()
+        attributionURLOpener = MockAttributionURLOpener()
+        mapView = MapView(
             frame: CGRect(origin: .zero, size: CGSize(width: 100, height: 100)),
             mapInitOptions: MapInitOptions(),
             dependencyProvider: dependencyProvider,
             orientationProvider: orientationProvider,
             urlOpener: attributionURLOpener)
+        window = UIWindow()
         window.addSubview(mapView)
-        notificationCenter.removeObserverStub.reset()
 
-        mapView.removeFromSuperview()
-
-        XCTAssertTrue(notificationCenter.removeObserverStub.invocations.contains(where: { invocation in
-            invocation.parameters.name == UIApplication.didEnterBackgroundNotification
-        }))
-        XCTAssertTrue(notificationCenter.removeObserverStub.invocations.contains(where: { invocation in
-            invocation.parameters.name == UIApplication.willEnterForegroundNotification
-        }))
-        XCTAssertTrue(notificationCenter.removeObserverStub.invocations.contains(where: { invocation in
-            invocation.parameters.name == UIScene.didEnterBackgroundNotification
-        }))
-        XCTAssertTrue(notificationCenter.removeObserverStub.invocations.contains(where: { invocation in
-            return invocation.parameters.name == UIScene.willEnterForegroundNotification
-        }))
-        XCTAssertEqual(notificationCenter.removeObserverStub.invocations.count, 4)
-    }
-
-    func testDisplayLinkPausedWhenAppMovingToBackground() {
-        notificationCenter.post(name: UIApplication.didEnterBackgroundNotification, object: nil)
+        metalView = try XCTUnwrap(dependencyProvider.makeMetalViewStub.invocations.first?.returnValue)
+        // reset is required here to ignore the setNeedsDisplay() invocation during initialization
+        metalView.setNeedsDisplayStub.reset()
 
-        XCTAssertEqual(displayLink.$isPaused.setStub.invocations.map(\.parameters), [true])
+        mapView.didMoveToWindow()
```

Comment:
```suggestion
```

What I was trying to say is that you don't need this line at all. `window.addSubview(mapView)` calls this method for you.

Reply:
that's true, good catch. I have removed it :-D